### PR TITLE
Introduce npm reference page

### PR DESCRIPTION
**Pull request by osmantoplica-trackunit**

### CODE REVIEW

File: sdk-docs/reference/npm-libraries.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+> 🚧 Beta
+> 
+> This is a beta version and subject to change without notice. Pricing, terms, conditions and availability may change in the final version.
+
+# npm Libraries
+
+The npm libraries that make up the SDK can be categorized into three types: vanilla JS libraries, tailwind-based styling libraries, and React libraries. You can browse all of our packages on [npmjs.com](https://www.npmjs.com/search?q=%40trackunit) too by searching for @trackunit.
+
+## Vanilla JS Libraries
+While we recommend building Iris Apps with React, the following libraries enable you to use whatever framework you want (even plain old js).
```
Comment (mmt-trackunit) [inline sdk-docs/reference/npm-libraries.md:15]:
```suggestion
While we recommend building Iris Apps with React, the following libraries enable you to use whatever framework you want (even plain Javascript).
```

File: sdk-docs/reference/npm-libraries.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+## Vanilla JS Libraries
+While we recommend building Iris Apps with React, the following libraries enable you to use whatever framework you want (even plain Javascript).
+
+| Library | Usage |
+| --- | --- |
+| [@trackunit/iris-app](https://www.npmjs.com/package/@trackunit/iris-app) | Core library for building Iris applications with any JS framework. It's a [Nx](https://nx.dev/)  plugin that adds helpful generators used to setup your app.|
+| [@trackunit/iris-app-api](https://www.npmjs.com/package/@trackunit/iris-app-api) | A small package containing reusable types. TODO: what is the difference between this and runtime-core-api?|
+| [@trackunit/iris-app-runtime-core](https://www.npmjs.com/package/@trackunit/iris-app-runtime-core) | TODO |
+| [@trackunit/iris-app-runtime-core-api](https://www.npmjs.com/package/@trackunit/iris-app-runtime-core-api) | A small package containing reusable types. TODO: what is the difference between this and iris-app-api? |
```
Comment (kla-trackunit) [inline sdk-docs/reference/npm-libraries.md:24]: @trackunit/iris-app-api -> Contains the types for the main parts of  Iris APP SDK, including manifest and so on.

@trackunit/iris-app-runtime-core -> Is the Library containg the code for the runtimes that are exposed to iris apps. These facilitate communication with the manager and other things like seting up the rest api.
Some runtimes might be part of other packages, this is the CORE once, that will be in all iris apps. 

@trackunit/iris-app-runtime-core-api -> Types for the @trackunit/iris-app-runtime-core package.

File: sdk-docs/reference/npm-libraries.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+| [@trackunit/css-component-tokens](https://www.npmjs.com/package/@trackunit/css-component-tokens) | Exposes the Trackunit Component Tokens Plugin. |
+| [@trackunit/css-tailwind](https://www.npmjs.com/package/@trackunit/css-tailwind) | Exposes the Trackunit Tailwind configuration. |
+| [@trackunit/css-tailwind-custom-properties-plugin](https://www.npmjs.com/package/@trackunit/css-tailwind-custom-properties-plugin) | Exposes a plugin used to convert the variables property in a Tailwind configuration to css custom properties. |
+| [@trackunit/tailwind-styled-components](https://www.npmjs.com/package/@trackunit/tailwind-styled-components) | Is a wrapper for [tailwind-styled-components](https://www.npmjs.com/package/tailwind-styled-components) Exposing a small utility for nested inline intellisense and some typing fixes. |
+| [@trackunit/ui-icons](https://www.npmjs.com/package/@trackunit/ui-icons) | Used by the Icon component in [@trackunit/react-components](https://www.npmjs.com/package/@trackunit/react-components) |
+| [@trackunit/ui-design-tokens](https://www.npmjs.com/package/@trackunit/ui-design-tokens) | TODO. |
```
Comment (kla-trackunit) [inline sdk-docs/reference/npm-libraries.md:44]: Exposes the design tokens used to generate the Tailwind Config, as well as a color utility used to get colors from the theme without using css variables.

File: sdk-docs/reference/npm-libraries.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+| Library | Usage |
+| --- | --- |
+| [@trackunit/react-core-hooks](https://www.npmjs.com/package/@trackunit/react-core-hooks) | Contains core React hooks used for getting data from our context providers inside Iris apps. |
+| [@trackunit/react-components](https://www.npmjs.com/package/@trackunit/react-components) | The home of Trackunits public UI library. |
+| [@trackunit/react-graphql-tools](https://www.npmjs.com/package/@trackunit/react-graphql-tools) | Provides tools for auto-generating hooks to interface our GraphQL API through React code. |
+| [@trackunit/react-core-contexts](https://www.npmjs.com/package/@trackunit/react-core-contexts) | Defines core context providers. TODO: brief description of what these providers are. |
+| [@trackunit/react-core-contexts-test](https://www.npmjs.com/package/@trackunit/react-core-contexts-test) | Contains testing utilities for [@trackunit/react-core-contexts](https://www.npmjs.com/package/@trackunit/react-core-contexts). |
+| [@trackunit/react-core-contexts-api](https://www.npmjs.com/package/@trackunit/react-core-contexts-api) | Contains reusable React types. TODO: elaborate slightly. |
```
Comment (kla-trackunit) [inline sdk-docs/reference/npm-libraries.md:55]: Holds to TrackunitProviders component, that is wrapping all iris apps by default. 
More info From the README: 

The TrackunitProviders component is exposed through the @trackunit/react-core-contexts package.
The component is a wrapper for all the base react providers needed to use contexts available in the trackunit manager.
This makes it possible to use Hooks and complex components in your Iris App.

Inside the provider the following hooks from react-core-hooks can be used:

    Get data on the current user through useCurrentUser hook.
    Read the current environment through the useEnvironment hook.
    Get the users subscription package and features with the useUserSubscription hook.
    Obtain an access token from the useToken hook.

File: sdk-docs/reference/npm-libraries.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+| Library | Usage |
+| --- | --- |
+| [@trackunit/react-core-hooks](https://www.npmjs.com/package/@trackunit/react-core-hooks) | Contains core React hooks used for getting data from our context providers inside Iris apps. |
+| [@trackunit/react-components](https://www.npmjs.com/package/@trackunit/react-components) | The home of Trackunits public UI library. |
+| [@trackunit/react-graphql-tools](https://www.npmjs.com/package/@trackunit/react-graphql-tools) | Provides tools for auto-generating hooks to interface our GraphQL API through React code. |
+| [@trackunit/react-core-contexts](https://www.npmjs.com/package/@trackunit/react-core-contexts) | Defines core context providers. TODO: brief description of what these providers are. |
+| [@trackunit/react-core-contexts-test](https://www.npmjs.com/package/@trackunit/react-core-contexts-test) | Contains testing utilities for [@trackunit/react-core-contexts](https://www.npmjs.com/package/@trackunit/react-core-contexts). |
```
Comment (kla-trackunit) [inline sdk-docs/reference/npm-libraries.md:54]: the -test is the test version of the providers, making it easier to test your apps. 
Link to the "How to test" page. (when it exists)

File: sdk-docs/reference/npm-libraries.md (hunk 4 above)
Comment (kla-trackunit) [inline sdk-docs/reference/npm-libraries.md:55]: Types fro the providers

Comment (osmantoplica-trackunit) [reply sdk-docs/reference/npm-libraries.md:56]: Could you elaborate?

Comment (kla-trackunit) [reply sdk-docs/reference/npm-libraries.md:56]: `react-core-contexts-api` holds the types for the providers and contexts in `react-core-contexts`.
This is what allows the use of the same hooks both inside the Iframe and Outside (this sentence is not relevant for 3. party i think)

Comment (kla-trackunit) [reply sdk-docs/reference/npm-libraries.md:56]: More info in this 2 min video: https://www.dropbox.com/s/d4cqil17yt40za3/Host%20Providers%20and%20App%20Providers%20%282min%29.mp4?dl=0

**Review by mta-trackunit — APPROVED** [review]

**Pull request merged**